docs/allocation.md

```diff
@@ -0,0 +1,152 @@
+
+## Heap sizing
+
+The overall size of the Java heap is determined by two command-line options, `-Xms`, which sets the initial size of the heap, and `-Xmx`, which sets the maximum size of the heap. Finer tuning of the heap depends on the heap configuration that is being used by a GC policy. For example, an LOA within the heap can be sized by using the [`-Xloainitial`, `-Xloaminimum`, and `-Xloamaximum`](xloaminimum.md) command-line options. A *nursery* area within the heap can be sized by using the [`-Xmn`, `-Xmns`, and `-Xmnx`](xmn.md) command-line options. For more information about policies and the heap configurations that are used, see [GC policies](gc.md). To determine the values that are in use for the Java heap, use the `-verbose:sizes` option when you run your Java application.
+
+### Expansion and shrinkage
+
+At startup, the VM allocates a single contiguous area of virtual storage to match the value of `-Xmx`. By default, this is 25% of the available memory up to a maximum of 25 GB. The actual Java heap size starts at the value set by `-Xms` and expands automatically, as required, up to the maximum heap size. The VM can also shrink the size of the Java heap. Expansion and shrinkage occur as part of a GC cycle when the VM has exclusive access to the heap. The only GC policy that does not support heap expansion and shrinkage is the `metronome` GC policy, where the heap is always fully expanded.
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i> **Note:** On operating systems that support paging, the VM allocates a single contiguous area that matches the value of `-Xms`. Additional memory is committed as the heap expands by using the paging process.
+
+Expansion occurs to maintain free space on the Java heap for object allocation. By default, the heap is expanded to maintain 30% free space, but this proportion can be adjusted by setting one of the following command-line options:
+
+- `-Xminf` determines the minimum proportion of the heap that must be free after garbage is removed.
+- `-Xmaxf` determines the maximum proportion of the heap that must be free after garbage is removed.
+
+If expansion is required, the amount of memory that the heap can expand by is controlled by the following command-line option:
+
+- `-Xmine` sets the minimum amount that the heap can expand by.
+- `-Xmaxe` sets the maximum amount that the heap can expand by. The default is unlimited expansion up to the maximum heap size (`-Xmx`).
+
+Expansion can also be triggered if more time is being spent on GC processing than is specified by the `-Xmaxt` option. In this case, the heap expands by an amount that provides 17% more free space, within the limits imposed by the `-Xmine` and `-Xmaxe` values.
+
```

Comment:
The above sets of options apply to:
- total heap in flat configuration (optthruput/optavgpause) and balanced
- only tenure area in generational heap (gencon)

Nursery (in gencon) resizing has it's own/rules and options to control (https://www.eclipse.org/openj9/docs/xgc/#dnssexpectedtimeratiomaximum). 

Eden (in balanced) has it's own rules , too. We are actively changing this behavior (both eden and total in balanced) - I'd wait to finish (in a couple of months) before documenting it.

Reply:
I've saved this comment to an issue we are working on to overhaul the options ... seemed appropriate if we are likely to have a change in behavior soon.